File: APIFramework/src/com/edu/qgMultiAPI/ShengchunyueTest.java
package com.edu.qgMultiAPI;

import java.io.IOException;

import org.apache.http.client.CookieStore;
import org.testng.annotations.Test;

import com.edu.common.Common;
import com.edu.core.HttpDriver;
import net.sf.json.JSONObject;

/*
 * @author=生春月
 */
public class ShengchunyueTest {

	String username = "200000000011";
	String password = "netease123";
	String loginUrl = "http://study-perf.qa.netease.com/common/fgadmin/login";
	String addressListUrl = "http://study-perf.qa.netease.com/fgadmin/address/list";
	String deleteUrl = "http://study-perf.qa.netease.com/fgadmin/address/delete";
	String newAddressUrl = "http://study-perf.qa.netease.com/fgadmin/address/new";
	String skuListUrl = "http://study-perf.qa.netease.com/common/skuList/";
	String getFeeUrl = "http://study-perf.qa.netease.com/common/getTransportFee?id=1&addressDetail=addressDetail";
	String submitUrl = "http://study-perf.qa.netease.com/fgadmin/orders/submit";

	@Test(description = "登录成功--查询收获地址--删除收获地址")
	public void scene1() throws IOException, Exception {
		// 登录成功
		JSONObject para = new JSONObject();
		para.element("phoneArea", "86");
		para.element("phoneNumber", username);
		para.element("password", password);
		String result1 = HttpDriver.doPost(loginUrl, para);
		System.out.println(result1);
		// 查询收获地址
		CookieStore cookie = Common.getLoginCookie(username, password);
		String result2 = HttpDriver.doGet(addressListUrl, cookie);
		System.out.println(result2);
		// 删除收获地址
		JSONObject json = JSONObject.fromObject(result2);
		JSONObject jresult = json.getJSONObject("result").getJSONArray("list").getJSONObject(0);
		int id = jresult.getInt("id");
		System.out.println(id);
		JSONObject del = new JSONObject();
		del.element("id", id);
		String result3 = HttpDriver.doPost(deleteUrl, del, cookie);
		System.out.println(result3);

	}

	@Test(description = "登录成功--查询收获地址--添加收获地址--删除收获地址")
	public void scene2() throws IOException, Exception {
		CookieStore cookie = Common.getLoginCookie(username, password);
		// 登录成功
		JSONObject para = new JSONObject();
		para.element("phoneArea", "86");
		para.element("phoneNumber", username);
		para.element("password", password);
		String result1 = HttpDriver.doPost(loginUrl, para);
		System.out.println(result1);
		// 添加收获地址
		JSONObject address = new JSONObject();
		address.element("id", "");
		address.element("receiverName", "李四");
		address.element("cellPhone", "200000000011");
		address.element("province", "江苏省");
		address.element("city", "南京市");
		address.element("area", "鼓楼区");
		address.element("addressDetail", "南京大学");
		String result2 = HttpDriver.doPost(newAddressUrl, address, cookie);
		System.out.println(result2);
		// 查询收获地址
		String result3 = HttpDriver.doGet(addressListUrl, cookie);
		System.out.println(result3);
		// 删除收获地址
		JSONObject json = JSONObject.fromObject(result3);
		JSONObject jresult = json.getJSONObject("result").getJSONArray("list").getJSONObject(0);
		int id = jresult.getInt("id");
		System.out.println(id);
		JSONObject del = new JSONObject();
		del.element("id", id);
		String result4 = HttpDriver.doPost(deleteUrl, del, cookie);
		System.out.println(result4);
	}

	@Test(description = "登录成功--获取所有商品的sku列表--查询收货地址--计算运费--提交订单")
	public void scene3() throws IOException, Exception {
		// 登录成功
		JSONObject para = new JSONObject();
		para.element("phoneArea", "86");
		para.element("phoneNumber", username);
		para.element("password", password);
		String result1 = HttpDriver.doPost(loginUrl, para);
		System.out.println(result1);
		// 获取所有商品的sku列表
		String result2 = HttpDriver.doGet(skuListUrl);
		System.out.println(result2);
		// 查询收获地址
		CookieStore cookie = Common.getLoginCookie(username, password);
		String result3 = HttpDriver.doGet(addressListUrl, cookie);
		System.out.println(result3);
		// 计算运费
		JSONObject json = JSONObject.fromObject(result3);
		JSONObject jresult = json.getJSONObject("result").getJSONArray("list").getJSONObject(0);
		String pro = jresult.getString("province");
		String city = jresult.getString("city");
		String area = jresult.getString("area");
		String addressDetail = pro + "_" + city + "_" + area;
		String result4 = HttpDriver.doGet(getFeeUrl);
		System.out.println(result4);
		// 提交订单
		JSONObject para2 = new JSONObject();
		para2.element("skuIds", "2,3");
		para2.element("skuNumbers", "1,1");
		para2.element("stockIds", "74966312,74966313");
		para2.element("receiverName", "张三");
		para2.element("cellPhone", "13245678986");
		para2.element("addressDetail", addressDetail);
		para2.element("province", pro);
		para2.element("city", city);
		para2.element("area", area);
		para2.element("voiceStatus", 0);
		para2.element("needInvoice", 0);
		para2.element("invoiceHead", "");
		para2.element("tranfportFee", 0);
		para2.element("lohisticsCompanyId", 1);
		para2.element("accessSource", "noSource");
		para2.element("accessDevice", 0);
		String result5 = HttpDriver.doPost(submitUrl, para2, cookie);
		System.out.println(result5);
	}

	@Test(description = "登录失败--查询收获地址--计算运费")
	public void scene4() throws IOException, Exception {
		// 登录失败
		JSONObject para = new JSONObject();
		para.element("phoneArea", "86");
		para.element("phoneNumber", username);
		para.element("password", password);
		String result1 = HttpDriver.doPost(loginUrl, para);
		System.out.println(result1);
		// 查询收获地址
		CookieStore cookie = Common.getLoginCookie(username, password);
		String result2 = HttpDriver.doGet(addressListUrl, cookie);
		System.out.println(result2);
		// 计算运费
		JSONObject json = JSONObject.fromObject(result2);
		JSONObject jresult = json.getJSONObject("result").getJSONArray("list").getJSONObject(0);
		String pro = jresult.getString("province");
		String city = jresult.getString("city");
		String area = jresult.getString("area");
		String addressDetail = pro + "_" + city + "_" + area;
		String result3 = HttpDriver.doGet(getFeeUrl);
		System.out.println(result3);
	}
}

File: Requests/QingGuo/MultiAPI/shengchunyue_test.py
import unittest
from core.Sendhttp import SendHttp
from core import Common


class qgSceneTest(unittest.TestCase):
    def setUp(self):
        self.LoginUrl = "/common/fgadmin/login"
        self.AddressListUrl = "/fgadmin/address/list"
        self.TransportFeeUrl = "/common/getTransportFee"
        self.SubmitUrl = "/fgadmin/orders/submit"
        self.Deleteurl = "/fgadmin/address/delete"
        self.Addurl = "/fgadmin/address/new"
        self.Skulisturl = "/common/skuList?goodsId=1&tm=1542093552569"

    # 登录成功--查询收获地址--删除收获地址
    def test_scene1(self):
        # 登录
        user = {"phoneArea": "86", "phoneNumber": "200000000044", "password": "netease123"}
        result = SendHttp().run_http(self.LoginUrl, "POST", user)
        self.assertEqual(result['code'], 200)
        self.assertEqual(result['message'], 'success')
        # 查询获取地址
        address_result = SendHttp().sent_get_bycookies(self.AddressListUrl, Common.getcookies(user))
        # 删除收获地址
        aid = address_result['result']['list'][0]['id']
        print(id)
        para = {"id": aid}
        delete_result = SendHttp().sent_post_bycookies(self.Deleteurl, Common.getcookies(user), para)

    # 登录成功--查询收获地址--添加收获地址--删除收获地址
    def test_scene2(self):
        # 登录
        user = {"phoneArea": "86", "phoneNumber": "200000000011", "password": "netease123"}
        result = SendHttp().run_http(self.LoginUrl, "POST", user)
        self.assertEqual(result['code'], 200)
        self.assertEqual(result['message'], 'success')
        # 查询获取地址
        address_result = SendHttp().sent_get_bycookies(self.AddressListUrl, Common.getcookies(user))
        pro = address_result['result']['list'][0]['province']
        city = address_result['result']['list'][0]['city']
        area = address_result['result']['list'][0]['area']
        # 添加收获地址
        address = {"id": "",
                   "receiverName": "李四",
                   "cellPhone": "200000000011",
                   "province": "江苏省",
                   "city": "南京市",
                   "area": "鼓楼区",
                   "addressDetail": "南京大学"}
        add_result = SendHttp().sent_post_bycookies(self.Addurl, Common.getcookies(user), address)
        # 删除收获地址
        id = address_result['result']['list'][0]['id']
        delete_result = SendHttp().sent_post_bycookies(self.Deleteurl, Common.getcookies(user), id)

    # 登录成功--获取所有商品的sku列表--查询收货地址--计算运费--提交订单
    def test_scene3(self):
        # 登录
        user = {"phoneArea": "86", "phoneNumber": "200000000011", "password": "netease123"}
        result = SendHttp().run_http(self.LoginUrl, "POST", user)
        self.assertEqual(result['code'], 200)
        self.assertEqual(result['message'], 'success')
        # 获取所有商品的sku列表
        sku_result = SendHttp().run_http(self.Skulisturl, "GET")
        # 查询获取地址
        address_result = SendHttp().sent_get_bycookies(self.AddressListUrl, Common.getcookies(user))
        pro = address_result['result']['list'][0]['province']
        city = address_result['result']['list'][0]['city']
        area = address_result['result']['list'][0]['area']
        # 计算运费
        addressDetails = address_result['result']['list'][0]['addressDetail']
        para = {"id": 1, "addressDetail": addressDetails}
        SendHttp().run_http(self.TransportFeeUrl, "GET", para)
        # 提交订单
        submit_data = {"skuIds": "1",
                       "skuNumbers": "1,1",
                       "stockIds": "74966312,74966313",
                       "receiverName": "张三",
                       "cellPhone": "200000000011",
                       "addressDetail": addressDetails,
                       "province": pro,
                       "city": city,
                       "area": area,
                       "voiceStatus": 0,
                       "needInvoice": 0,
                       "invoiceHead": "",
                       "transportFee": 0,
                       "logisticsCompanyId": 1,
                       "accessSource": "noSource",
                       "accessDevice": 0}
        SendHttp().sent_get_bycookies(self.AddressListUrl, Common.getcookies(user), submit_data)

    # 登录失败--查询收获地址--计算运费
    def test_scene4(self):
        # 登录失败
        user = {"phoneArea": "86", "phoneNumber": "200000000011", "password": "netease12345"}
        result = SendHttp().run_http(self.LoginUrl, "POST", user)
        self.assertEqual(result['code'], 400)
        self.assertEqual(result['message'], '用户名或者密码错误')
        # 查询获取地址
        address_result = SendHttp().sent_get_bycookies(self.AddressListUrl, Common.getcookies(user))
        # 计算运费
        try:
            addressDetails = address_result['result']['list'][0]['addressDetail']
            para = {"id": 1, "addressDetail": addressDetails}
            SendHttp().run_http(self.TransportFeeUrl, "GET", para)
        except KeyError:
            print("获取运费失败")


if __name__ == '__main__':
    unittest.main()
